CMakeLists.txt

```diff
@@ -1,4 +1,9 @@
 cmake_minimum_required(VERSION 3.15)
+cmake_policy(
+    SET
+    CMP0091
+    NEW
+    )
```

Comment:
What does this do? ^^"

Reply:
Cmake is backwards compatible even through breaking changes by seeing the requires cmake version and setting certain policies.

This policy is for how the runtime library flag is being resolved and defaults to OLD
we explicitly want the NEW behavior because the OLD behavior caches the values and could create potentially incorrect results.

See https://cmake.org/cmake/help/latest/policy/CMP0009.html